services/wallet/activity/filter.go

```diff
@@ -85,49 +86,137 @@ type Filter struct {
 	FilterOutCollectibles bool    `json:"filterOutCollectibles"`
 }
 
-func GetRecipients(ctx context.Context, db *sql.DB, offset int, limit int) (addresses []eth.Address, hasMore bool, err error) {
-	rows, err := db.QueryContext(ctx, `
+func GetRecipients(ctx context.Context, db *sql.DB, chainIDs []common.ChainID, addresses []eth.Address, offset int, limit int) (recipients []eth.Address, hasMore bool, err error) {
+	queryFormatString := `
+	WITH filter_conditions AS (
```

Comment:
It will be nice to externalize the big queries in separate `.sql` files, as we do with the activity filter, to increase the readability.